cpp/src/parquet/column_writer.cc

```diff
@@ -1708,7 +1728,7 @@ Status TypedColumnWriterImpl<DType>::WriteArrowDictionary(
                       AddIfNotNull(rep_levels, offset));
     std::shared_ptr<Array> writeable_indices =
         indices->Slice(value_offset, batch_num_spaced_values);
-    if (page_statistics_) {
+    if (page_statistics_ || bloom_filter_) {
```

Comment:
The part of code is for dictionary. Previously, `page_statistics` will call `update_stats` to get dictionary items used in dictionary, and use them for "min-max".

Here I reuse the code the part, and making it update the bloom-filter using `referenced_dictionary`.